gammapy/datasets/spectrum.py

```diff
@@ -237,6 +237,51 @@ def plot_residuals_spatial(self, *args, **kwargs):
     def to_spectrum_dataset(self, *args, **kwargs):
         raise NotImplementedError("Already a Spectrum Dataset. Method not supported")
 
+    def get_resampled_energy_axis(self, name="energy", conditions={}):
+        """Returns an energy axis whose binning satisfies given conditions on the per-bin statistics.
+
+        Parameters
+        ----------
+        name: str
+            Name of the axis to be resampled.
+        conditions : dict
+            Keyword arguments containing the per-bin conditions used to resample the axis.
+            Available options are: "min_counts", "min_excess".
+
+        Returns
+        -------
+        energy_axis : `~gammapy.maps.MapAxis`
+            New energy axis.
+        """
+        available_conditions = ["min_counts", "min_excess"]
+
+        def _find_idx(key):
+            if key in available_conditions:
+                data = getattr(self, key.strip("min_")).data[::-1]
+                cumsum_idx = np.where(np.cumsum(data) >= conditions[key])[0][0]
```

Comment:
We already have https://docs.gammapy.org/0.19/api/gammapy.maps.Map.html#gammapy.maps.Map.cumsum, which returns a new `Map` object with correctly adapted axis definition. This might help to simplify the implementation here.